lib/filewriter.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * FileWriter -- fast non-streaming file appender for accepting buffered data
+ *
+ * Locks the file when appending to be safe with multiple writers.
+ * Also, it reopens the file periodically, to make it work with logrotate.
+ *
+ * Copyright (C) 2014-2015 Andras Radics
+ * Licensed under the Apache License, Version 2.0
+ */
+
+var fs = require('fs');
+var fse = require('./fs-ext.js');
+
+var flock = fse.flock;
+var flockSync = function( fd, mode ) {
+    try { fse.flockSync(fd, mode); return true }
+    catch (err) { return false }
```

Comment:
right now flockSync is only used to unlock fd's, but I'll change it to return `true` or the error.